packages/fast-components-react-base/src/tabs/tabs.tsx

```diff
@@ -0,0 +1,336 @@
+            this.getSlot(TabSlot.tabItem)
+        );
+        const currentItemIndex: number = items.findIndex(this.getCurrentIndexById);
+        const nextItemIndex: number = currentItemIndex < items.length - 1 ? currentItemIndex + 1 : 0;
+        const nextItemId: string = items[nextItemIndex].props.id;
+
+        if (!this.props.activeId) {
+            this.setState({
+                activeId: nextItemId
+            });
+
+            (Array.from(this.tabListRef.current.children)[nextItemIndex] as HTMLButtonElement).focus();
+        } else if (typeof this.props.onUpdateTab === "function") {
+            this.props.onUpdateTab(nextItemId);
+        }
+    }
+
+    /**
+     * Activates the first tab item
+     */
+    private activateFirst(): void {
+        const items: JSX.Element[] = this.getChildBySlot(
+            this.props.children,
+            this.getSlot(TabSlot.tabItem)
+        );
+        const activeId: string = items[0].props.id;
+
+        if (!this.props.activeId) {
+            this.setState({
+                activeId
+            });
+
+            (Array.from(this.tabListRef.current.children)[0] as HTMLButtonElement).focus();
+        } else if (typeof this.props.onUpdateTab === "function") {
+            this.props.onUpdateTab(activeId);
+        }
+    }
+
+    /**
+     * Activates the last tab item
+     */
+    private activateLast(): void {
+        const items: JSX.Element[] = this.getChildBySlot(
+            this.props.children,
+            this.getSlot(TabSlot.tabItem)
+        );
+        const lastItemIndex: number = items.length - 1;
+        const activeId: string = items[lastItemIndex].props.id;
+
+        if (!this.props.activeId) {
+            this.setState({
+                activeId
+            });
+
+            (Array.from(this.tabListRef.current.children)[lastItemIndex] as HTMLButtonElement).focus();
+        } else if (typeof this.props.onUpdateTab === "function") {
+            this.props.onUpdateTab(activeId);
+        }
+    }
+
+    /**
+     * Gets the current index by tab item ID
+     */
+    private getCurrentIndexById = (item: JSX.Element): boolean => {
+        return item.props.id === this.state.activeId;
+    }
+
+    /**
+     * Gets the slot to use for tab children
+     */
+    private getSlot(slot: TabSlot): TabSlot | string {
+        switch (slot) {
+            case TabSlot.tab:
+                return typeof this.props.tabSlot === "string" ? this.props.tabSlot : TabSlot.tab;
+            case TabSlot.tabItem:
+                return typeof this.props.tabItemSlot === "string" ? this.props.tabItemSlot : TabSlot.tabItem;
+            case TabSlot.tabPanel:
+                return typeof this.props.tabPanelSlot === "string" ? this.props.tabPanelSlot : TabSlot.tabPanel;
+        }
+    }
+
+    /**
+     * Gets the child by the slot property
+     */
+    private getChildBySlot(children: React.ReactNode, slot: TabSlot | string): JSX.Element[] {
```

Comment:
This returns a collection of children, correct? I'd recommend changing the function name to reflect that - this makes it sound like this function will return a single child.